✨ Add :[Mission 1~4] 완료, 코드 refactor 완료 / 추가기능 구현 중

hjkdw95 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
- `feature/hyunjaeKim` branch 생성
- 페이지 스타일링 최적화
- Mission 1 - 사용자 입력 데이터 저장 기능 구현
- Mission 2 - 로그인 버튼 활성화 기능 구현 완료
- Mission 3 - 댓글 기능 구현 완료
- Mission 4 - 댓글 컴포넌트화, props로 데이터 전달 구현
- `추가 구현` - Article 컴포넌트 화 및 Article 댓글 관련 Mock data 추가
- `추가 구현` - 각 Article 별 댓글 추가 기능 구현
- `추가 구현` - 각 Article 별 댓글 삭제 기능 구현
- `추가 구현` - 회원가입 페이지 구현
- `Refactor` - 각종 중첩 사항 정리, 모든 요소 component화 완료
- `Refactor` - DB 받아오는 경로 변경 및 state 처리 방법 변경

		 
## 수정 사항들 자세한 내용
- `feature/hyunjaeKim` branch 생성
- 기존 local에 저장해두었던 scss 파일 현재 repository환경에 최적화
- `Login` : `event.target.value`로 입력 값 전달받아 setState로 업데이트하여 구현 완료 (id, pw 각각 이벤트 함수 생성 - 보완 필요)
- `Login`: 로그인 유효성 인증 로직에 `onChange`이벤트, 삼항연산자 활용
- `Login`: 회원가입 페이지 구현 및 route 추가
- `Comment` : form 태그에 `onSubmit`이벤트를 활용하여 댓글입력 후 엔터 및 버튼 클릭 시 댓글 list up 되도록 함
- `Comment` : 댓글 입력 창 별도로 `Comment` Component 생성하여 분리
- `Comment` : 댓글 입력 시 Submit 버튼 활성화 구현 및 `Replies`창에 업데이트 되도록 구현 완료
- `Comment` : 댓글 입력, 삭제에 관련된 내용을 각 feed별 state 생성하여 해당 state에서 관리되도록 함
- `Replies` : 댓글 나타내는 창 별도로 Component 생성하여 분리
- `Reply` : 댓글 1개에 대한 Component 생성하여 분리
- `Reply` :  `json`파일 받아와 각 피드별로 댓글 업데이트 가능하도록 구현 (입력도 가능)
- `All` : 모든 컴포넌트 요소 컴포넌트 화 완료


## 기타 질문 및 특이 사항
추가 구현 예정 사항
- 댓글 내용에 `@`가 추가되는 경우에만 서식 변경
- `content`부분 좀 더 세부적으로 componen화 예정

질문
- 연결할 contents가 없는 상태에서 `link`태그를 사용해도 괜찮은지 궁굼합니다. 혹, contents가 없는 상황에서라도 `link`태그를 사용해야 한다면, `link`태그를 사용하기 위해 빈 `component`를 일일이 만들어야 하는지 궁굼합니다.

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

hjkdw95 committed:
💄 Add : feature/hyunjaeKim 브랜치 생성 및 UI 스타일링 최적화

hjkdw95 committed:
🎨 Modify : Aside 관련 파일 경로 수정

hjkdw95 committed:
🔥 Modify: navbar 관련 내용 main.js에서 삭제

hjkdw95 committed:
✨ Add : [Mission 1] Login | 사용자 입력 데이터 저장 완료
- event.target.value로 입력 값 전달받아 setState로 state업데이트하여 구현 완료

hjkdw95 committed:
✨ Add : [Mission 2] Login | 로그인 버튼 활성화 (validation)
-  ID - @ 포함 / PW - 5글자 이상 이상인 경우에만 로그인 버튼 색상 활성화
- 삼항연산자 및 state 활용하여 구현

hjkdw95 committed:
✨ Add : [Mission 4] Main | 댓글 컴포넌트화 + props로 데이터 전달

hjkdw95 committed:
🎨 [Mission: 3] Main | 댓글 기능 구현 및 댓글 입력 칸 컴포넌트 화
- form onSubmit이벤트를 활용하여 댓글입력 후 엔터 및 버튼 클릭 시 댓글 list up 되도록 함
- 댓글 입력 시 Submit 버튼 활성화 구현
- 댓글 입력 form만 별도로 comment component로 export

hjkdw95 committed:
👍 Merge : branch 'main' into feature/hyunjaeKim

hjkdw95 committed:
🎨 Add : Nav Component 추가
- merge 후 중복된 css 서식 삭제
- Nav Component Main.js에 추가

hjkdw95 committed:
💄 Modify : profileImg border 및 background size 수정

hjkdw95 committed:
💄 Add : Fontawesome link 추가

hjkdw95 committed:
💄  Add : fontawesome 추가

hjkdw95 committed:
♻️ Refactor: Article 컴포넌트 화

hjkdw95 committed:
✨ Add : contents  component 구현
- map 사용하여 mock data 각 아티클에 연결

hjkdw95 committed:
🤡 Add : Article 관련 Mock Data 추가

hjkdw95 committed:
🎨 Add : 각 피드별 댓글 추가 기능 구현
- map으로 전체 feed를 돌려, id 값이 일치하는 내역만 반영되도록 작업

hjkdw95 committed:
♻️ Refactor: 구조분해 할당 적용

hjkdw95 committed:
💄 Add : login input form border 서식 추가

hjkdw95 committed:
♻️ Refactor: login validation 함수 refactoring
- PW, ID 별 분리되어있던 함수 1 개로 통합

hjkdw95 committed:
♻️ Refactor: import 순서 정정

orosy commented on src/pages/hyunjaeKim/Login/Login.js:
```diff
@@ -1,8 +1,62 @@
+      isActive:
+        this.state.userID.includes("@") && this.state.userPW.length > 4
+          ? true
+          : false,
```
삼항연산자가 아주 아름답네요🎨

hjkdw95 replied:
감사합니다^____^

orosy commented on src/pages/hyunjaeKim/Login/Login.js:
```diff
@@ -1,8 +1,62 @@
+            type="email"
+            className="userID"
+            placeholder="전화번호, 사용자 이름 또는 이메일"
+            aria-label="Write account ID"
```
aria-label 이라고 값을 주신 건 혹시 뭔가요? 이것도 Semantic한 건가요 ㅇ_ㅇ? (궁금)

dudgus5766 replied:
저도 궁금합니다!

hjkdw95 replied:
네 시맨틱이기도 하고, SEO관련된 것인데요, `aria-label`을 달아야 시각장애인용 웹리더기에서 내용을 읽을 수 있다고 하더라고요, 그래서 가급적 정보성이 짙은 요소들은 라벨을 달려고 하고 있습니다!

dudgus5766 replied:
와 시각장애인용 웹리더기는 처음 들어보네요..정보 감사합니다!

orosy commented on src/pages/hyunjaeKim/Main/Aside/Aside.scss:
```diff
@@ -0,0 +1,84 @@
+
+.suggestProfileDetails {
+  display: flex;
+  flex-direction: column;
```
저는 flex-direction 아직 능숙하게 사용 못하는데 멋있네유👍👍

hjkdw95 replied:
감사합니다:)

orosy commented on src/pages/hyunjaeKim/Main/Main.js:
```diff
@@ -1,8 +1,155 @@
+
+  handleAdd = (comment, content) => {
+    const commentList = [
+      ...content.comments,
```
이것이 바로 댓글 추가.. 잠시 눈이 부셔서 선글라스 좀 쓸게요😎

orosy commented on src/pages/hyunjaeKim/Main/Replies/Comment.js:
```diff
@@ -0,0 +1,51 @@
+  };
+
+  formRef = React.createRef();
+  inputRef = React.createRef();
```
createRef는 아직 저에게는 어려운 개념이네요.. 멋있습니다 현재님👍

hjkdw95 replied:
아니에요..태수님도 금방 사용하실 거여서...숙쓰럽습니다^^

orosy commented on src/pages/hyunjaeKim/Main/styles/searchBar.scss:
```diff
@@ -0,0 +1,49 @@
+  width: 16px;
+  height: 16px;
+  transform: rotate(45deg);
+  box-shadow: 0 0 5px 1px rgba(var(--jb7, 0, 0, 0), 0.0975);
```
이건 저번에도 놀랐지만 이런 생각을 하시다니! 대단대단

hjkdw95 replied:
구글링..이랄까요~~?ㅎㅎ...

orosy commented on src/pages/hyunjaeKim/Main/styles/storyFeed.scss:
```diff
@@ -0,0 +1,29 @@
+  justify-content: flex-start;
+  align-items: center;
+  padding: 16px 10px;
+  overflow: hidden;
```
overflow hidden이랑 auto가 다른 건가요?? 🤔

hjkdw95 replied:
네 auto는 길어지면 스크롤바가 생기는 것으로 아는데..hidden은 일정 길이가 넘어가도 스크롤바가 생기지 않습니다!

orosy commented on src/pages/hyunjaeKim/stylesComponent/base/layouts.scss:
```diff
@@ -0,0 +1,47 @@
+  margin-bottom: 20px;
+}
+
+@media screen and (max-width: 1000px) {
```
크으 반응형👍

orosy commented on src/pages/hyunjaeKim/stylesComponent/modules/profileImg.scss:
```diff
@@ -0,0 +1,49 @@
+    height: 36px;
+    margin: 0;
+    margin-right: 10px;
+    background: linear-gradient(to bottom right, #f58529 20%, #c42d91);
```
linear-gradient라는 건 또 처음 보네요..! 마지막까지 눈이 부셔서 선글라스를 벗을 수가 없네요😎

hjkdw95 replied:
ㅎㅎ아닙니다 좋게 봐주셔서 감사합니다

orosy reviewed (commented):
일주일간 위스타그램으로 달리시느라 고생 많이 하셨습니다 현재님!
코드 리뷰 드리니 확인해주시면 감사하겠습니다🐸

dudgus5766 commented on public/data/commentData.json:
```diff
@@ -0,0 +1,66 @@
+      {
+        "id": 3,
+        "userName": "영현님",
+        "comment": "동희님이 싫어요!",
```
이건 진심이 아닙니다ㅠㅠ

hjkdw95 replied:
어머 이게 왜들어갔죠...!! 철진니임~~!!!!

dudgus5766 commented on src/pages/hyunjaeKim/Login/Login.js:
```diff
@@ -1,8 +1,62 @@
+    this.setState({
+      [className]: value,
+      isActive:
+        this.state.userID.includes("@") && this.state.userPW.length > 4
```
삼항연사자를 이런식으로 쓰니 훨씬 보기 좋네요! 저도 적용해보겠습니다🙂

dudgus5766 commented on src/pages/hyunjaeKim/Main/Aside/Aside.js:
```diff
@@ -0,0 +1,168 @@
+import React, { Component } from "react";
```
`Aside.js`가 이런 용도였군요..저만 비어있네요

hjkdw95 replied:
이제 채우시면 되죠^0^

dudgus5766 commented on src/pages/hyunjaeKim/Main/Aside/Aside.scss:
```diff
@@ -0,0 +1,84 @@
+  float: left;
+
+  &::after {
+    content: "\00B7";
```
혹시 li after로 `content: "\00B7"` 이 용도가 어떤 것인지 알 수 있을까요?

hjkdw95 replied:
저게 가운데 찍히는 점을 유니코드로 입력한 것입니다~! (About * Help ...)

dudgus5766 commented on src/pages/hyunjaeKim/Main/Contents/Contents.js:
```diff
@@ -0,0 +1,15 @@
+
+export default class Contents extends Component {
+  render() {
+    const { feeds, onSubmit } = this.props;
```
구조 분해 할당, 이런 식으로 사용하셨군요!
알려주셔서 감사합니다🙂

dudgus5766 commented on src/pages/hyunjaeKim/Main/Main.scss:
```diff
@@ -1 +1,17 @@
-//main
+/* base */
```
scss 파일을 따로따로 각주를 두고 분리해놓으셨네요🤯 생각지도 못한..이런 식으로 분리해놓으니 어디서 어떤 scss 파일을 받고 있는 지 한 눈에 정확하게 파악이 가능하겠네요! 👍

hjkdw95 replied:
넵 컴포넌트화 할때도 잘 꺠지지 않는답니다^^

saengmotmi replied:
@hjkdw95 이러한 사용도 나쁘지 않아보이는데 저는 개인적으로 컴포넌트의 뼈대와 로직, 스타일이 서로 인접한 곳에서 관리되었으면 싶습니다. 만약 Main 컴포넌트라면 해당 컴포넌트 상단에 './index.scss'를 정의할 것 같다는 뜻입니다. 어떻게 생각하세요??

hjkdw95 replied:
네 javascript 한 파일로 관리하던 때랑 다르게 react는 컴포넌트 별로 관리하다보니까 stylesheet도 각 react component에 맞추어 관리하는것이 유지 보수에 보다 관리에 용이하다는 것을 체감하고 있습니다.
본 사항 참고하여 추후 프로젝트 진행 시 각 react component별로 stylesheet가 관리되도록 개선해보겠습니다:)

dudgus5766 commented on src/pages/hyunjaeKim/Main/Replies/Comment.js:
```diff
@@ -0,0 +1,51 @@
+  onSubmit = (event, content) => {
+    event.preventDefault();
+    const comment = this.inputRef.current.value;
+    comment && this.props.onSubmit(comment, content);
```
if 표현식을 조건부 렌더링 &&으로 표현하니 코드가 짧아지고 명료해지네요👍

hjkdw95 replied:
넵 조건 연산자가 좋더라구요

dudgus5766 commented on src/pages/hyunjaeKim/stylesComponent/base/fonts.scss:
```diff
@@ -0,0 +1,89 @@
+/****************** Fonts ******************/
```
scss도 리액트에서 컴포넌트 나누듯이 나눈 건 정말 충격 🤯
저도 다음 프로젝트에서는 꼭 한 번 써보겠습니다!

hjkdw95 replied:
넵 응원합니다!

dudgus5766 reviewed (commented):
현재님! 이번 위스타그램 너무 수고하셨습니다. 하나하나 알려주셔서 너무너무 감사해요🙂 
코드 리뷰 남겼습니다! 확인부탁드립니다 ㅎㅎ

JUCHEOLJIN commented on src/pages/hyunjaeKim/Main/Replies/Comment.js:
```diff
@@ -0,0 +1,51 @@
+        />
+        <button
+          type="submit"
+          className={
```
옆집에서 놀러왔습니다. 동적인 클래스네임 너무 잘 사용하신 것 같아요! 많이 배우고 갑니다!👍 ✨

JUCHEOLJIN reviewed (commented):
현재님에게 항상 많이 배웁니다.
열정적으로 하시는 모습보고 옆에서 더 힘차게 하는 것 같아요! 
프로젝트까지 화이팅입니다!! ✨

JUCHEOLJIN commented on src/pages/hyunjaeKim/stylesComponent/modules/forms.scss:
```diff
@@ -0,0 +1,51 @@
+  font-size: 14px;
+  line-height: 18px;
+  color: #a5a7aa;
+}
```
큰 요소들을 css로 컴포넌트처럼 분리해서 작성하신 부분이 인상깊습니다. 
css 를 찾아볼 때 시간이 많이 걸리는 편인데, 금방 찾을 수 있어서 유지보수의 큰 장점이 될 것 같습니다. ✨

hjkdw95 replied:
좋게봐주셔서 감사합니다:)

hjkdw95 committed:
🔥 Remove : 지워진 파일 정리

hjkdw95 committed:
💄 Modify: myProfle class 명 currentProfile로 변경

hjkdw95 committed:
♻️ Refactor: Story feed component화

hjkdw95 committed:
♻️ Refactor: Aside 부분 컴포넌트화

hjkdw95 committed:
♻️ Refactor: MainArea부분 컴포넌트화

hjkdw95 committed:
🎨 Modify: 디렉토리 구조 변경

hjkdw95 committed:
✨ Add: 회원가입 component 추가
- API 연결 test

hjkdw95 committed:
♻️ Refactor : article 요소 component 화

hjkdw95 committed:
✨ Add : 각 feed 별 댓글 삭제 기능 추가

hjkdw95 committed:
🐛 Modify: 임시로 추가했던 routes 각주처리
- 지워도 됩니다

hjkdw95 committed:
♻️ Refactor: feed별로 state관리되도록 refactor완료

hjkdw95 committed:
♻️ Refactor : state lifecycle로 받아오도록 변경
- componentDidMount, setState 활용하여 state 직접적으로 변경되는 것 방지

hjkdw95 committed:
🐛 Fix : key props 생략되었던 부분 추가..

hjkdw95 committed:
♻️ Refactor : content 데이터 수신 범위 변경
- feed object가 한겹 벗겨지도록 구조화

hjkdw95 committed:
✨ Add: 하트 버튼 클릭 시 색상 변경 기능 추가

hjkdw95 committed:
🔥 Remove : spread operator 삭제

hjkdw95 committed:
✨ Add : 하트 버튼 클릭에 따라 likes 증감 구현

hjkdw95 committed:
🐛 Fix : handleAdd 메소드 spread operator 경로 정정

saengmotmi commented on public/data/commentData.json:
```diff
@@ -0,0 +1,66 @@
+        "id": 1,
+        "userName": "현재님",
+        "comment": "안녕하세요!",
+        "isUser": false
```
`isUser`라는 프로퍼티는 혹시 댓글 작성자가 자기 자신인지를 판별하기 위한 목적으로 추가해주신 건가요?

hjkdw95 replied:
넵 맞습니다. 자기 자신인지 판단해서 본인이 작성한 댓글만 삭제되려는 기능을 구현하려고 되었는데 진행이 늦어졌습니다!
본인이 작성한 댓글만 삭제되는 기능 금일로 추가 구현 완료되어 참고 부탁드립니다:)

saengmotmi commented on src/pages/hyunjaeKim/Login/Login.js:
```diff
@@ -1,8 +1,80 @@
+      method: "POST",
+      body: JSON.stringify({
+        ...this.state,
+      }),
```
```suggestion
      body: JSON.stringify({ email, password }),
```
스프레드 연산자를 잘 활용해주셨으나 제 생각에는 state에 들어가는 내용이 꼭 body에 필요한 프로퍼티와 일치하지 않을 수 있기 때문에 state 전체를 넣어주는 건 부적절해보이네요. 어떤 속성들이 들어가는지 명시해주되 구조 분해 할당과 객체 단축 속성명 문법을 사용해주면 조금 더 간결하게 표현할 수 있을 것 같습니다.

hjkdw95 replied:
넵 필요한 부분만 구조분해할당하여 간결하게 정정하였습니다:)!

saengmotmi commented on src/pages/hyunjaeKim/Login/Login.js:
```diff
@@ -1,8 +1,80 @@
+  handlePWInput = e => {
+    const { value } = e.target;
+    this.setState({ ...this.state, password: value });
+  };
```
```suggestion
  handleInput = e => {
    const { name, value } = e.target;
    this.setState({ ...this.state, [name]: value });
  };
```
- 기능 : input 태그로부터 value를 넘겨 받아 setState
- 대상 : email, password state

코드를 살펴보면 두 메서드의 기능은 동일하나 업데이트 하는 대상이 달라짐에 따라 중복 로직을 추가로 작성하신 것 같습니다.
input 태그의 name 속성과 객체 계산된 속성명 문법을 사용하면 위와 같이 업데이트 로직에 관련된 메서드를 하나로 줄일 수 있습니다.

hjkdw95 replied:
넵! 로그인 백프론트 결합 시 날로 작업한 내역이 정리가 안되어있었네요ㅠㅠ! 제안해주신 방법으로 개선하였습니다✨

saengmotmi commented on src/pages/hyunjaeKim/Login/Login.js:
```diff
@@ -1,8 +1,80 @@
+          />
+          <button
+            type="button"
+            className={this.state.isActive ? "loginBtn active" : "loginBtn"}
```
```suggestion
            className={`loginBtn ${isActive ? 'active' : ''}`}
```
구조분해 할당과 템플릿 리터럴을 활용하면 위와 같은 표현도 가능합니다. 고정된 요소만 놔두고 변화하는 요소만 바꿔주면 중복을 피할 수 있겠네요.

hjkdw95 replied:
넵! 반영하여 변경하였습니다:)

saengmotmi commented on src/pages/hyunjaeKim/Login/LoginBox/LoginBox.scss:
```diff
@@ -0,0 +1,16 @@
+  border: 1.5px solid #e6e6e6;
+  background-color: #fff;
+  text-align: center;
+}
```
요소를 가운데 정렬하는 많은 방법들 중에 fixed + top, left 선택하신 이유가 궁금합니다!
이 경우 요소가 가운데로 오기만 하면 돼서 fixed를 쓰더라도 외관상 문제는 없어보이나 만약 요소가 추가 된다면 각 요소들 간의 순서나 상대적 위치를 잡기 힘들어 질 것 같다는 의견입니다.

hjkdw95 replied:
넵 위치를 가운데로 잡으려고 단순히 fixed를 사용했습니다! 요소 추가할 때 순서 및 사이즈 변동 고려하여 다시 변경하였습니다:)

saengmotmi replied:
@hjkdw95 넵 가급적 일반적인 경우 poistion 속성 대신 block / inline 속성 사용해서 잡아주세요. 그래야 수정에 유연한 레이아웃이 될 것 같습니다.

saengmotmi commented on src/pages/hyunjaeKim/Login/makeAccount.js:
```diff
@@ -0,0 +1,135 @@
+      method: "POST",
+      body: JSON.stringify({
+        ...this.state,
+      }),
```
여기도 위와 비슷한 의견입니다

hjkdw95 replied:
state 내의 6개 key에 대한 value를 모두 전달해야했어서 최초에 위와 같이 기재했었습니다! 혹 후에 추가적으로 state 내역이 추가될 가능성도 있기에 본 건 또한 구조분해할당 하여 전송되도록 수정하였습니다:)

saengmotmi commented on src/pages/hyunjaeKim/Login/makeAccount.js:
```diff
@@ -0,0 +1,135 @@
+  handlePWInput = e => {
+    const { value } = e.target;
+    this.setState({ ...this.state, password: value });
+  };
```
이 함수들도 동일한 로직을 공유하고 있고 위에 남겨드렸던 리뷰 참고하셔서 줄여보시면 좋을 것 같습니다.

hjkdw95 replied:
넵 참고하여 로직 정정하였습니다:)

saengmotmi commented on src/pages/hyunjaeKim/Main/Aside/Aside.scss:
```diff
@@ -0,0 +1,84 @@
+  justify-content: flex-start;
+  align-items: center;
+  margin: 18px 0 22px 0;
+
```
가장 바깥쪽 요소(`.sideBar`)의 셀렉터로 셀렉터 전체를 nesting 해주면 css에 스코프가 생기면서 각 스타일 들이 서로 간섭하지 않게 될 거 같은데 어떻게 생각하시나요?

hjkdw95 replied:
제가 요령이 없는지 nesting후 공통사항 소거하면 스타일이 무너집니다..우선 nesting 반영 및 flexbox mixin 활용하여 개선하였습니다:)

saengmotmi commented on src/pages/hyunjaeKim/Main/MainArea/Contents/Content.js:
```diff
@@ -0,0 +1,39 @@
+import DetailWrapper from "./DetailWrapper";
+
+export default class Content extends Component {
+  state = {};
```
- content는 부모에서 받아온 값을 초기값으로
- comments는 값이 존재할 것임에도 불구하고 초기값 설정이 안되어 있어서
이 두 값에 대해서 알맞은 자료형으로 초기값 설정해주시면 좋겠습니다

hjkdw95 replied:
넵 초기값 설정 완료하였습니다

saengmotmi commented on src/pages/hyunjaeKim/Main/MainArea/Contents/Contents.js:
```diff
@@ -0,0 +1,31 @@
+  componentDidMount() {
+    fetch("http://localhost:3000/data/commentData.json", {
+      method: "GET",
+    })
```
```suggestion
    fetch("/data/commentData.json")
```
이렇게 사용해도 동작은 같습니다

hjkdw95 replied:
넵 정정해주신 사항 반영하였습니다:)!

saengmotmi commented on src/pages/hyunjaeKim/Main/MainArea/Contents/DetailWrapper.js:
```diff
@@ -0,0 +1,73 @@
+    this.props.onDelete(reply);
+  };
+
+  handleBtn = () => {
+    this.setState({ ...this.state, isRed: !this.state.isRed }, () => {
+      this.handleIncrement();
+    });
+  };
```
이렇게 callback으로 작성하면 렌더가 두번 돌게 되는데, 꼭 두번 돌려야 하는 로직인가요?

hjkdw95 replied:
네 처음에 버튼 색 변경과 likes 수 증가를 동시에 state에 업데이트 시키려고 했는데, 그렇게 하니까 setState함수 특성 상 비동기적으로 렌더가 되어서 화면 상에 하트 색은 변경이 안되고 likes 수만 변경되는 것으로 확인되었습니다.
억지로 비동기를 끊기 위해서 부득이하게 callback을 사용하였습니다.

saengmotmi commented on src/pages/hyunjaeKim/Main/MainArea/Contents/DetailWrapper.js:
```diff
@@ -0,0 +1,73 @@
+    this.state.isRed === true
+      ? this.setState({ ...this.state, likes: this.state.likes + 1 })
+      : this.setState({ ...this.state, likes: this.state.likes - 1 });
+  };
```
```suggestion
handleIncrement = () => {
  const { isRed, likes } = this.state
  
  const sign = { true: 1, false: -1 }
  
  this.setState((prev) => ({
    ...prev,
    likes: likes + sign[isRed]
  }))  
};
```
- 함수 내 데이터가 외부 값을 바라보기 보다는 최대한 순수하게 함수 내부의 값을 사용 (ex. this.state -> prev)
- this.setState는 등은 갇은 동작이라면 처음 데이터가 들어가는 거겠죠.

hjkdw95 replied:
넵 함수 내 데이터는 최대한 함수 내부의 값을 사용하도록 주의하겠습니다! 동 건도 반영 완료입니다:)

saengmotmi commented on src/pages/hyunjaeKim/Main/MainArea/Contents/Replies/Comment.js:
```diff
@@ -0,0 +1,49 @@
+    event.preventDefault();
+    const comment = this.inputRef.current.value;
+    comment && this.props.onSubmit(comment);
+    this.formRef.current.reset();
```
if문이 더 좋아보입니다. 이유는 다음과 같습니다.
- comment 이후 &&의 연산 결과로 떨어진 값을 사용하는 경우가 아님 (문장과 표현식의 차이)
- 만약 참 거짓 판단 후 처리해야 하는 로직이 2줄 이상이 된다면 결국 if 문으로 수정해야 함

그리고 comment 유무에 무관하게 reset이 실행되어야 하나요?

hjkdw95 replied:
넵 참고하여 if문으로 변경하였습니다.

comment의 경우 value가 들어가 있는 경우만 reset이 필요합니다!

saengmotmi commented on src/pages/hyunjaeKim/Main/MainArea/Contents/Replies/Comment.js:
```diff
@@ -0,0 +1,49 @@
+
+  handleButton = () => {
+    const comment = this.inputRef.current.value;
+    this.setState({ isActive: comment.length > 0 ? true : false });
```
```suggestion
    this.setState({ isActive: comment.length > 0 });
```
동일한 표현입니다

hjkdw95 replied:
넵 본 사항 참고하여 정정하였습니다.

saengmotmi commented on src/pages/hyunjaeKim/Main/MainArea/Contents/Replies/Replies.js:
```diff
@@ -0,0 +1,25 @@
+            onDelete={this.handleDelete}
+            userName={reply.userName}
+            comment={reply.comment}
+            key={reply.id}
```
어차피 reply 전체를 props로 전달하는데 굳이 별도의 props로 분리하여 정의할 필요 없어보입니다

hjkdw95 replied:
넵 동 건 반영하여 정정 완료입니다:)

saengmotmi commented on src/pages/hyunjaeKim/Main/MainArea/StoryFeed/StoryFeed.js:
```diff
@@ -0,0 +1,31 @@
+      { id: 6, userID: "chewhee_l..." },
+      { id: 7, userID: "0oo0o0y" },
+      { id: 8, userID: "euddeme_" },
+    ],
```
실제로는 비동기로 처리되어야 하는 값들이기 때문에 state에 초기값으로 넣어놓기보다는 public에 json 형태로 빼두고 호출하는 편이 더 실제와 가까워보입니다

hjkdw95 replied:
넵 별도로 json 만들어서 받아오도록 정정하였습니다

saengmotmi commented on src/pages/hyunjaeKim/Main/styles/searchBar.scss:
```diff
@@ -0,0 +1,49 @@
+}
+
+.searchResultBox {
+  width: 375px;
+  height: 60px;
```
form 전체 뿐만 아니라 각각의 요소들에 대해서도 고정 크기를 주고 계신데, 가급적이면 고정 크기는 피하는 것이 좋습니다. 부모 요소의 크기는 자식 요소의 크기 + margin 등으로 결정됩니다. 이 부분 생각해보시고 자식 요소로부터 부모 크기가 결정 될 수 있도록 짜보세요!

hjkdw95 replied:
넵 동 건 금일 개인 serachBar componnet 추가 시 반영되도록 하겠습니다:)

hjkdw95 replied:
```css
  .searchComponent {
    position: relative;
    max-height: 350px;
    background-color: #fff;
    border-radius: 6px;
    overflow: auto;
    z-index: 1;
  }

  .searchResultBox {
    width: 375px;
    height: 60px;
  }
```
넵 위와 같이 자식 요소의 사이즈만 따르도록 정정하였습니다!

saengmotmi commented on src/pages/hyunjaeKim/stylesComponent/modules/profileImg.scss:
```diff
@@ -0,0 +1,49 @@
+/*? 이렇게 클래스로 따로 주는게 더 좋은가 */
+.focus {
+  background-color: #000;
+}
```
이렇게 간단한 네이밍의 클래스가 nesting 없이 바깥으로 빠지게 되면 앱 전체에 영향을 주는 속성이 됩니다. 여타 부분도 마찬가지로 특별한 이유 없다면 컴포넌트 전체를 감싸는 셀렉터 하나 사용하셔서 감싸주세요.

hjkdw95 replied:
네 연관된 요소에 포함시켰습니다!

saengmotmi reviewed (commented):
코드가 깔끔해서 보기 좋았습니다. 리뷰 확인해주세요!

hjkdw95 committed:
♻️ Refactor : Login Page 중복 로직 제거

hjkdw95 committed:
♻️ Refactor: loginContainer 가운데 정렬 방법 변경
- container 감싸주는 div추가하여 flexbox활용

hjkdw95 committed:
♻️ Refactor: fetch JSON 전송 내역 수정
- 모든 state 내용을 전달하는 것에서 필요한 내역만 구조분해할당 통하여 전달되도록 정정

hjkdw95 committed:
♻️ Refactor: 중첩 로직 개선
-input 이벤트 관련 중첩된 로직을 1개 로직에서 모두 컨트롤 가능하도록 개선

hjkdw95 committed:
♻️ Refactor : Aside nesting 및 flexbox mixin 사용

hjkdw95 committed:
🐛 Fix: sidebar flex direction 수정
- space-betweenv -> column

hjkdw95 committed:
♻️ Refactor: state 기본값 설정

hjkdw95 committed:
♻️ Refactor: fetch url 개선
- localhost 부분 제거

hjkdw95 committed:
♻️ Refactor: storyfeed data -> json파일로 분리

hjkdw95 committed:
♻️ Refactor: isActive 삼항연산자 개선

hjkdw95 committed:
♻️ Refactor: reply props 개선
- 중첩된 props를 받아오는 것으로 확인되어 reply 하나만 props로 받아오도록 개선

hjkdw95 committed:
♻️ Refactor: onSubmit메소드 조건문 if으로 변경
- comment 이후 &&의 연산 결과로 떨어진 값을 사용하는 경우가 아님 (문장과 표현식의 차이)

hjkdw95 committed:
♻️ Refactor: handleIncrement 함수 개선
- 함수내에서 직접적으로 state를 건드려 수정하지 않도록 정정

hjkdw95 committed:
♻️ Refactor: focus class nesting 적용

hjkdw95 committed:
🐛 Fix: Loginbox  import경로 수정

hjkdw95 committed:
🎨 Add : 본인 댓글만 삭제되도록 로직 추가 구현

hjkdw95 committed:
♻️ Refactor: Navbar 재구성 및 component화

hjkdw95 committed:
✨ Add : 검색기능 테스트 용 목데이터 제작

hjkdw95 committed:
✨ Add : 검색 기능 구현 완료
- componentDidUpdate, filter 활용

hjkdw95 committed:
🐛 Fix : 입력값 없을 시 검색결과 창 보이지 않도록 수정

hjkdw95 committed:
🎨 Add : NavMenu component 추가

hjkdw95 committed:
🩹 Modify: searchData object명 변경

hjkdw95 committed:
🔥 Delete: searchBar.scss 삭제

hjkdw95 committed:
🐛  Fix: SearchBar 충돌된 css 보완

hjkdw95 committed:
✨ Add: Navmenu toggle 기능 추가

hjkdw95 committed:
🩹 Fix : container margin 크기 조정

hjkdw95 committed:
🔥 Delete : previewData 삭제

hjkdw95 committed:
✨ Add : text-overflow 효과 추가
- to storyfeed ID and detailwrapper user comment

hjkdw95 committed:
✨ Add: more버튼 클릭 시 본문 내용 펼쳐지도록 기능 추가

hjkdw95 committed:
🐛 Fix: searchData lifecycle method 제거
- componentDidUpdated 사용 시 계속 render가 일어나 네트워크 통신 과부화 확인됨
- onFocus 이벤트 활용하여 focus된 후에 data 받아오는 방식으로 변경

hjkdw95 committed:
🎨 Add: 댓글 중 @가 포함된 댓글만 서식 변경
- hashtag class 입혀서 색상 변경

hjkdw95 committed:
🚧 Add : storyfeed 버튼 클릭 시 translate 작동 작업 중

hjkdw95 committed:
🩹 Modify: writter comment관련 css 파일 변경

hjkdw95 committed:
✨ Add : storyfeed carousel기능 추가

hjkdw95 committed:
🎨 Modify: userID fullname으로 정정
